src/java/generator/MapGenerator.java

```diff
@@ -170,6 +172,30 @@ public SCMap generate(long seed) throws ExecutionException, InterruptedException
 			noProps.fillCircle(map.getHydros()[i].x, map.getHydros()[i].z, 7, true);
 		}
 
+		WreckGenerator wreckGenerator = new WreckGenerator(map, random.nextLong());
+		BinaryMask t1LandWreckMask = new BinaryMask(64, random.nextLong());
+		t1LandWreckMask.randomize(0.01f).intersect(grass.getBinaryMask()).minus(noProps).deflate(1);
+		BinaryMask t2LandWreckMask = new BinaryMask(64, random.nextLong());
+		t2LandWreckMask.randomize(0.005f).intersect(grass.getBinaryMask()).minus(noProps).minus(t1LandWreckMask);
+		BinaryMask t3LandWreckMask = new BinaryMask(64, random.nextLong());
+		t3LandWreckMask.randomize(0.01f).intersect(grass.getBinaryMask()).minus(noProps).minus(t1LandWreckMask).minus(t2LandWreckMask).trimEdge(128);
+		BinaryMask t2NavyWreckMask = new BinaryMask(64, random.nextLong());
+		BinaryMask navyLandCopy = new BinaryMask(land.getBinaryMask(), random.nextLong());
+		t2NavyWreckMask.randomize(0.01f).intersect(navyLandCopy.outline()).minus(noProps);
+		BinaryMask navyFactoryWreckMask = new BinaryMask(64, random.nextLong());
+		navyFactoryWreckMask.randomize(0.01f).minus(grass.getBinaryMask()).minus(noProps).deflate(2).trimEdge(10);
+		UnitGenerator unitGenerator = new UnitGenerator(map, random.nextLong());
+
+		wreckGenerator.generateWrecks(t1LandWreckMask, wreckGenerator.T1_Land, 2f);
+		wreckGenerator.generateWrecks(t2LandWreckMask, wreckGenerator.T2_Land, 15f);
+		wreckGenerator.generateWrecks(t3LandWreckMask, wreckGenerator.T3_Land, 30f);
+		wreckGenerator.generateWrecks(t2NavyWreckMask, wreckGenerator.T2_Navy, 60f);
+		wreckGenerator.generateWrecks(navyFactoryWreckMask, wreckGenerator.Navy_Factory, 120f);
+
+		BinaryMask allWreckMask = new BinaryMask(513, random.nextLong());
+		allWreckMask.combine(t1LandWreckMask).combine(t2LandWreckMask).combine(t3LandWreckMask).combine(t2NavyWreckMask).inflate(2);
+		noProps.combine(allWreckMask);
```

Comment:
Extracting that into a method would be really cool :D

Reply:
I can put this into a method but had left it out since all the other prop and mask generation is done directly in the generate function and this generation does rely on a few few of those masks. I could also clean up the ordering as well.